File: Assignment40/program4.cpp
/////////////////////////////////////////////////////////////////// 
//
// File name : program4.cpp
// Description : Pattern printing using recursion
// Author : Adinath Santosh Pawar
// Date : 21/7/25
// 
///////////////////////////////////////////////////////////////////

#include<iostream>
using namespace std;

void Display()
{
    static char ch = 'A';

    if(ch <= 'F')
    {
        cout<<ch<<" ";
        ch++;
        Display();
    }
}


int main()
{
    Display();

    return 0;
}

File: Assignment49/program4.java
import java.util.*;

/////////////////////////////////////////////////////////////////// 
//
// File name : program4.java
// Description : Pattern printing
// Author : Adinath Santosh Pawar
// Date : 21/8/25           
// 
///////////////////////////////////////////////////////////////////

class Pattern
{
    public void Pattern(String str)
    {
        for(int i = 0; i < str.length(); i++)
        {
            for(int j = 0; j < str.length(); j++)
            {
                if(i + j < str.length())
                {
                    System.out.print(str.charAt(j) + " ");
                }
                else
                {
                    System.out.print("* ");
                }
            }
            System.out.println();
        }

        for(int i = 1; i < str.length(); i++)   // i from 1 to avoid duplicate at centre
        {
            for(int j = 0; j < str.length(); j++)
            {
                if(i >= j)
                {
                    System.out.print(str.charAt(j) + " ");
                }
                else
                {
                    System.out.print("* ");
                }
            }
            System.out.println();
        }
    }
}

class program4
{
    public static void main(String A[])
    {
       Pattern pobj = new Pattern();
       Scanner sobj = new Scanner(System.in);

       System.out.print("Enter the string ");
       String str = sobj.nextLine();

       pobj.Pattern(str);
    }
}